ext/sockets/sockets.c

```diff
@@ -1911,15 +1940,23 @@ PHP_FUNCTION(socket_set_option)
 				RETURN_THROWS();
 			}
 
-			convert_to_long(sec);
-			convert_to_long(usec);
+			zend_long zsec = zval_try_get_long(sec, &failed);
+			if (failed) {
+				zend_argument_type_error(4, "\"%s\" must be of type int, %s given", sec_key, zend_zval_value_name(sec));
+				RETURN_THROWS();
+			}
+			zend_long zusec = zval_try_get_long(usec, &failed);
+			if (failed) {
+				zend_argument_type_error(4, "\"%s\" must be of type int, %s given", usec_key, zend_zval_value_name(usec));
+				RETURN_THROWS();
+			}
 #ifndef PHP_WIN32
-			tv.tv_sec = Z_LVAL_P(sec);
-			tv.tv_usec = Z_LVAL_P(usec);
+			tv.tv_sec = zsec;
+			tv.tv_usec = zusec;
 			optlen = sizeof(tv);
 			opt_ptr = &tv;
 #else
-			timeout = Z_LVAL_P(sec) * 1000 + Z_LVAL_P(usec) / 1000;
+			timeout = zsec * 1000 + zusec / 1000;
```

Comment:
I think I've missed that in the other PR: `timeout` is `DWORD` (that is 32bit unsigned), so the assignment might cause wrap-around. I think we should catch that.

Reply:
> I think I've missed that in the other PR: `timeout` is `DWORD` (that is 32bit unsigned), so the assignment might cause wrap-around. I think we should catch that.

DWORD on master tough here still int but issue can definitively occur very much.

Reply:
Oops, I've mixed that up! If this targets PHP-8.3, we cannot change `convert_to_long()` to use `zval_try_get_long()` (too much BC break). Should better use `zval_get_long()` instead.